pkg/health/condition/check_all_members.go

```diff
@@ -35,19 +35,26 @@ func (a *allMembersReady) Check(_ context.Context, etcd druidv1alpha1.Etcd) Resu
 
 	result := &result{
 		conType: druidv1alpha1.ConditionTypeAllMembersReady,
-		status:  druidv1alpha1.ConditionTrue,
-		reason:  "AllMembersReady",
-		message: "All members are ready",
+		status:  druidv1alpha1.ConditionFalse,
+		reason:  "NotAllMembersReady",
+		message: "At least one member is not ready",
 	}
 
-	for _, member := range etcd.Status.Members {
-		if member.Status != druidv1alpha1.EtcdMemberStatusReady {
-			result.status = druidv1alpha1.ConditionFalse
-			result.reason = "NotAllMembersReady"
-			result.message = "At least one member is not ready"
+	notAllMembersRegistered := int32(len(etcd.Status.Members)) < etcd.Spec.Replicas
+	if notAllMembersRegistered {
+		return result
+	}
```

Comment:
```suggestion
	if int32(len(etcd.Status.Members)) < etcd.Spec.Replicas {
	        // not all members are registered yet
		return result
	}
```

Reply:
only did this for better readability. But if you think this is not necessary and the condition is self-evident then i can accept the change.

Reply:
Condition seems self-evident, and extra variable just increases memory usage. Just a small code comment should be fine. If you wish, you can also introduce a new condition reason like:
```
result.reason = "NotAllMembersRegistered"
result.message = "At least one member is not registered"
```

But maybe this is unnecessary.